### PR TITLE
Get fee recommendation from mempool space

### LINKED ISSUE #1274: Switch to mempool.space fee estimation

### PR DESCRIPTION
**Pull request by ercecan**
# Description
Mempool space fee estimation

## Linked Issues
- Fixes #1274

### CODE REVIEW

**Review by kpp — APPROVED** [review]
Lgtm but maybe this url should be set via config?

File: crates/bitcoin-da/src/service.rs [hunk 1]
```diff
@@ -56,6 +56,9 @@ use crate::REVEAL_OUTPUT_AMOUNT;
 pub const FINALITY_DEPTH: u64 = 8; // blocks
 const POLLING_INTERVAL: u64 = 10; // seconds
 
+const MEMPOOL_SPACE_RECOMMENDED_FEE_ENDPOINT: &str =
+    "https://mempool.space/testnet4/api/v1/fees/recommended";
+
```
Comment (yaziciahmet) [inline crates/bitcoin-da/src/service.rs:61]: IMO this should be set via config and it should be optional. If the url is not passed in config, then estimate_smart_fee should still be used

Comment (eyusufatik) [inline crates/bitcoin-da/src/service.rs:61]: you should construct the url with the network field in daservice

**Review by eyusufatik — CHANGES_REQUESTED** [review]

File: crates/bitcoin-da/src/service.rs (hunk 1 above)
Comment (ercecan) [reply crates/bitcoin-da/src/service.rs]: talked with @eyusufatik and decided it was not necessary, check the new impl after last commit

**Review by eyusufatik — APPROVED** [review]

**Pull request merged**